api/python/quilt3/packages.py

```diff
@@ -851,20 +852,33 @@ def set_meta(self, meta):
         return self
 
     def _fix_sha256(self):
-        entries = [entry for key, entry in self.walk() if entry.hash is None]
-        if not entries:
-            return
+        """
+        Calculate and set missing hash values
+        """
+        def _fix_pkg_entries(entries):
+            physical_keys = []
+            sizes = []
+            for entry in entries:
+                physical_keys.append(entry.physical_key)
+                sizes.append(entry.size)
 
-        physical_keys = []
-        sizes = []
-        for entry in entries:
-            physical_keys.append(entry.physical_key)
-            sizes.append(entry.size)
+            results = calculate_sha256(physical_keys, sizes)
+
+            entries_w_missing_hash = []
+            for entry, obj_hash in zip(entries, results):
+                if obj_hash is None:
+                    entries_w_missing_hash.append(entry)
+                else:
+                    entry.hash = dict(type='SHA256', value=obj_hash) if obj_hash is not None else None
```

Comment:
`obj_hash is not None` is always true for this branch, so

```suggestion
                    entry.hash = dict(type='SHA256', value=obj_hash)
```

Reply:
Why is `obj_hash` never equal to `None`? In this branch, `calculate_sha256` can return empty (`None`) hash values.

Reply:
OK, I see what you mean now.